compiler/optimizer/VPHandlers.cpp

```diff
@@ -5966,7 +6082,19 @@ TR::Node *constrainAdd(OMR::ValuePropagation *vp, TR::Node *node)
    if (rhs)
       {
       constraint = NULL;
-      if (rhs->asLongConst())
+      if (rhs->asFloatConst())
+         {
+         if (rhs->asFloatConst()->getFloat() > FLOAT_NEG_INFINITY &&
+             rhs->asFloatConst()->getFloat() < FLOAT_POS_INFINITY)
+            constraint = TR::VPEqual::create(vp, rhs->asFloatConst()->getFloat());
+         }
+      else if (rhs->asDoubleConst())
+         {
+         if (rhs->asDoubleConst()->getDouble() > FLOAT_NEG_INFINITY &&
+             rhs->asDoubleConst()->getDouble() < FLOAT_POS_INFINITY)
```

Comment:
These 2 lines look like typos. Should you be using the DOUBLE* versions of infinity ?